specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -12,821 +12,329 @@
-          (this.mappings?.lines ?? []).length === 0 &&
-          hasPermission('/workbench/dataset', 'update')
-        ) {
-          const dialog = this.options.display(
-            <Dialog
-              buttons={
-                <>
-                  <Button.DialogClose>{commonText.close()}</Button.DialogClose>
-                  <Link.Info
-                    href={`/specify/workbench/plan/${this.dataset.id}/`}
-                  >
-                    {commonText.create()}
-                  </Link.Info>
-                </>
-              }
-              header={wbPlanText.noUploadPlan()}
-              onClose={() => dialog()}
-            >
-              {wbPlanText.noUploadPlanDescription()}
-            </Dialog>
-          );
-          this.$('.wb-validate, .wb-data-check')
-            .prop('disabled', true)
-            .prop('title', wbText.wbValidateUnavailable());
-        } else {
-          this.$('.wb-validate, .wb-data-check').prop('disabled', false);
-          this.$('.wb-show-upload-view')
-            .prop('disabled', false)
-            .prop('title', undefined);
-        }
-
-        /*
-         * These methods update HOT's cells settings, which resets meta data
-         * Thus, need to run them first
-         */
-        configureHandsontable(
-          this.hot!,
-          this.mappings,
-          this.dataset,
-          pickLists
-        );
-
-        if (this.dataset.rowresults) this.validation.getValidationResults();
-
-        // CHeck if any column is reordered
-        if (this.dataset.visualorder?.some((column, index) => column !== index))
-          this.hot?.updateSettings({
-            manualColumnMove: writable(this.dataset.visualorder),
-          });
-
-        this.wbUtils.findLocalityColumns();
-        this.wbUtils.render();
-
-        this.hotIsReady = true;
-
-        this.hotCommentsContainer = (document.getElementsByClassName(
-          'htComments'
-        )[0] ?? undefined) as HTMLElement | undefined;
-      });
+        : dataset.rows,
+    [dataset]
+  );
+
+  const [hotTable, setHotTable] = React.useState<HotTable | null>(null);
+  const hot = hotTable?.hotInstance ?? undefined;
+
+  const isUploaded = React.useMemo<boolean>(
+    () => dataset.uploadresult !== null && dataset.uploadresult.success,
+    [dataset]
+  );
+
+  const mappings = React.useMemo(
+    (): WbMapping | undefined => parseWbMappings(dataset),
+    [dataset]
+  );
+
+  const throttleRate = Math.ceil(clamp(10, data.length / 10, 2000));
+
+  const [hasUnsavedChanges, spreadsheetChanged, spreadsheetUpToDate] =
+    useBooleanState();
+
+  const [cellCounts, setCellCounts] = React.useState<WbCellCounts>({
+    newCells: 0,
+    invalidCells: 0,
+    searchResults: 0,
+    modifiedCells: 0,
+  });
+
+  const workbench = React.useMemo<Workbench>(() => {
+    const workbench: Workbench = {
+      data,
+      dataset,
+      hot: hot,
+      mappings: mappings!,
+      throttleRate,
```

Comment:
(optional) remove throttleRate from here and instead create a util for computing `throttleRate` given dataset rows count and use that util where needed - that would help decouple this large object